src/mlpack/core/data/split_data.hpp

```diff
@@ -368,6 +341,241 @@ Split(const arma::Mat<T>& input,
+ */
+template <class FieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename FieldType::object_type>::value ||
+              arma::is_Mat_only<typename FieldType::object_type>::value>>
+void Split(const FieldType& input,
+           FieldType& trainData,
+           FieldType& testData,
+           const double testRatio,
+           const bool shuffleData = true)
+{
+  const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
+  const size_t trainSize = input.n_cols - testSize;
+
+  trainData.set_size(1, trainSize);
+  testData.set_size(1, testSize);
+
+  arma::uvec order = arma::linspace<arma::uvec>(0, input.n_cols - 1,
+      input.n_cols);
+  if (shuffleData)
+    order = arma::shuffle(order);
+
+  if (trainSize > 0)
+  {
+    for (size_t i = 0; i < trainSize; i++)
+       trainData[i] = input(0, order(i));
+  }
+
+  if (testSize <= input.n_cols)
+  {
+    for (size_t i = trainSize; i < input.n_cols - 1; ++i)
+       testData[i - trainSize] = input(0, order(i));
+  }
+}
+
+/**
+ * Given an input dataset and labels, split into a training set and test set.
+ * Example usage below.  This overload returns the split dataset as a std::tuple
+ * with four elements: an FieldType containing the training data, an
+ * FieldType containing the test data, an arma::field<arma::vec> containing the
+ * training labels, and an arma::field<arma::vec> containing the test labels.
+ *
+ * The input dataset must be of type arma::field. It should have the shape -
+ * (n_rows = 1, n_cols = Number of samples, n_slices = 1)
+ *
+ * NOTE: Here FieldType could be arma::field<arma::mat> or arma::field<arma::vec>
+ *
+ * @code
+ * arma::field<arma::mat> input = loadData();
+ * arma::field<arma::vec> label = loadLabel();
+ * auto splitResult = Split(input, label, 0.2);
+ * @endcode
+ *
+ * @param input Input dataset to split.
+ * @param inputLabel Input labels to split.
+ * @param testRatio Percentage of dataset to use for test set (between 0 and 1).
+ * @param shuffleData If true, the sample order is shuffled; otherwise, each
+ *       sample is visited in linear order. (Default true).
+ * @return std::tuple containing trainData (FieldType), testData
+ *      (FieldType), trainLabel (arma::field<arma::vec>), and
+ *                   testLabel (arma::field<arma::vec>).
+ */
+template <class FieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename FieldType::object_type>::value ||
+              arma::is_Mat_only<typename FieldType::object_type>::value>>
+std::tuple<FieldType, FieldType, FieldType, FieldType>
+Split(FieldType& input,
+      arma::field<arma::vec>& inputLabel,
+      const double testRatio,
+      const bool shuffleData = true)
+{
+  FieldType trainData;
+  FieldType testData;
+  arma::field<arma::vec> trainLabel;
+  arma::field<arma::vec> testLabel;
+
+  Split(input, inputLabel, trainData, testData, trainLabel, testLabel,
+    testRatio, shuffleData);
```

Comment:
```suggestion
      testRatio, shuffleData);
```
Just a spacing fix. :)